### ISSUE TITLE
Concurrent create_distributed_table and master_remove_node fails 

### ISSUE THREAD

**Issue by velioglu** [citusdata/citus#1397]
To reproduce : 

- Connection 1
```
create table tt1(id int);
begin;
    select create_distributed_table('tt1', 'id');
```

- Connection 2
` select master_remove_node('host-name', port);`

- Connection 1
`end;`

```
Insert into tt1 values(5);
ERROR:  failed to assign 1 task(s) to worker nodes
```

**Comment by metdos** [comment]
@velioglu, could you also add the expected behavior?

**Comment by velioglu** [comment]
`master_remove_node` supposed to delete nodes without any active shard on it only. Otherwise, it deletes the tuple from `pg_dist_node` with the given node-name and node port, but shard placements with the same name and port are still there in the `pg_dist_shard` table. So, the expected behaviour would be preventing `master_remove_node` from removing the node.

**Comment by metdos** [comment]
The fix here is simple actually, getting a lock on `pg_dist_shard` in master_remove_node() solves the problem.

**Comment by sumedhpathak** [comment]
@metdos Are we planning to fix this in 6.2 then? Thanks.

**Comment by metdos** [comment]
> @metdos Are we planning to fix this in 6.2 then? Thanks.

Not necessarily. After talking with @ozgune, we inclined to not prioritize it. However, if we would like to fix it, I can create the PR and have somebody review it tomorrow.

**Comment by marcocitus** [comment]
Note that this has been fixed on master, though it might be worth adding some isolation tests.
